apps/ca.c

```diff
@@ -518,12 +516,10 @@ int ca_main(int argc, char **argv)
     /*****************************************************************/
     /* we definitely need a private key, so let's get it */
 
-    if ((keyfile == NULL) && ((keyfile = NCONF_get_string(conf,
-                                                          section,
-                                                          ENV_PRIVATE_KEY)) ==
-                              NULL)) {
-        lookup_fail(section, ENV_PRIVATE_KEY);
-        goto end;
+    if (keyfile == NULL) {
+        keyfile = lookup_conf(conf, section, ENV_PRIVATE_KEY);
+        if (keyfile == NULL)
+            goto end;
```

Comment:
It really is a matter of taste...  however, I agree with @richsalz and would rather like to see this:

``` C
    if (keyfile == NULL
        && (keyfile = lookup_conf(conf, section, ENV_PRIVATE_KEY)) == NULL)
        goto end;
```